<launch> 
    <arg name = "namespace" default = "mushr"/>
    <arg name = "goal_config" default = "3.0, 3.0, 0.0"/>
    <arg name = "planning_cycle_duration" default = "0.5"/>
    <arg name = "max_cycles" default="30"/>
    <arg name = "preprocess_timeout" default = "0.05" />
    <arg name = "postprocess_timeout" default = "0.1" />
    <arg name = "visualize" default = "false"/>
    <arg name = "record" default = "false"/>
    <arg name = "params_file" default="examples/replanning/obstacle_0.yaml" />
    <group ns="$(arg namespace)">
        <node pkg="task_planning" name="replanner" type="$(arg namespace)_replanning_dirt" output="screen">
        <!-- <node pkg="task_planning" name="replanner" type="$(arg namespace)_replanning_rogue" output="screen"> -->
            <param name="goal_config" value="$(arg goal_config)" />
            <param name="planning_cycle_duration" value="$(arg planning_cycle_duration)" />
            <param name="preprocess_timeout" value="$(arg preprocess_timeout)" />
            <param name="postprocess_timeout" value="$(arg postprocess_timeout)" />
            <param name="max_cycles" value="$(arg max_cycles)" />
            <param name="visualize" value="$(arg visualize)" />
            <param name="params_file" value="$(arg params_file)" />
        </node>
        <node if="$(arg record)" pkg="rosbag" name="rosbag_record" type="record" 
        args="-O $(find task_planning)/data/$(arg namespace).bag 
        /mushr/control /mushr/plan_stepper/control /mushr/ml4kp_plan 
        /mushr/ml4kp_traj /mushr/pose /mushr/aruco_cTw/robot_pose"/>
    </group>
</launch>